File: frontend/src/types/ApiConnection.ts
export interface ApiConnection {
  id: number;
  userId: number;
  apiProvider: "GitHub" | "Spotify" | "Discord";
  accessToken: string;
  refreshToken: string;
  tokenExpiresAt: string;
  lastSyncAt: string;
  isActive: boolean;
  settings: string;
}

export interface ApiConnectionStatus {
  provider: "GitHub" | "Spotify" | "Discord";
  isConnected: boolean;
  lastSync?: string;
}

File: backend/PersonalTimelineAPI/Models/ApiConnection.cs
namespace PersonalTimelineAPI.Models
{
    public class ApiConnection
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string ApiProvider { get; set; } = string.Empty; 
        public string AccessToken { get; set; } = string.Empty;
        public string RefreshToken { get; set; } = string.Empty;
        public DateTime TokenExpiresAt { get; set; }
        public DateTime LastSyncAt { get; set; } = DateTime.UtcNow;
        public bool IsActive { get; set; } = true;
        public string Settings { get; set; } = "{}"; 
        
        // Navigation property
        public User? User { get; set; }
    }
}